tutorial.md

```diff
@@ -0,0 +1,1022 @@
+windows.  Naively, we can achieve this by spawning a pipe and letting
+xmonad feed the relevant information to that pipe.  The problem with
+that approach is that when the pipe is not being read and gets full,
+xmonad will freeze!
+
+It is thus much better to switch over to property based logging, where
+we are writing to an X11 property and having xmobar read that; no danger
+when things are not being read!  For this reason we have to use
+`XMonadLog` instead of `StdinReader` in our xmobar.  There's also an
+`UnsafeXMonadLog` available, should you want to send actions to xmobar
+sometimes (this is useful, for example, for
+[XMonad.Util.ClickableWorkspaces], which is a new feature in `0.17`).
+
+_IF YOU ARE ON A VERSION `< 0.17`_: As discussed above, the `xmobar`
+  function uses pipes, so you actually do want to use the `StdinReader`.
+  Simply replace _all_ occurences of `XMonadLog` with `StdinReader`
+  below (don't forget the template!)
+
+## Configuring Xmobar
+
+Now, before this will work, we have to configure xmobar.  Here's a nice
+starting point.
+
+``` haskell
+  Config { overrideRedirect = False
+         , font     = "xft:iosevka-9"
+         , bgColor  = "#5f5f5f"
+         , fgColor  = "#f8f8f2"
+         , position = TopW L 90
+         , commands = [ Run Weather "EGPF"
+                          [ "--template", "<weather> <tempC>°C"
+                          , "-L", "0"
+                          , "-H", "25"
+                          , "--low"   , "lightblue"
+                          , "--normal", "#f8f8f2"
+                          , "--high"  , "red"
+                          ] 36000
+                      , Run Cpu
+                          [ "-L", "3"
+                          , "-H", "50"
+                          , "--high"  , "red"
+                          , "--normal", "green"
+                          ] 10
+                      , Run Alsa "default" "Master"
+                          [ "--template", "<volumestatus>"
+                          , "--suffix"  , "True"
+                          , "--"
+                          , "--on", ""
+                          ]
+                      , Run Memory ["--template", "Mem: <usedratio>%"] 10
+                      , Run Swap [] 10
+                      , Run Date "%a %Y-%m-%d <fc=#8be9fd>%H:%M</fc>" "date" 10
+                      , Run XMonadLog
+                      ]
+         , sepChar  = "%"
+         , alignSep = "}{"
+         , template = "%XMonadLog% }{ %alsa:default:Master% | %cpu% | %memory% * %swap% | %EGPF% | %date% "
+         }
+```
+
+First, we set the font to use for the bar, as well as the colors.  The
+position options are documented well on the [xmobar home page] or,
+alternatively, in the [quick-start.org] on GitHub.  The particular
+option of `TopW L 90` says to put the bar in the upper left of the
+screen, and make it consume 90% of the width of the screen (we need to
+leave a little bit of space for `trayer-srg`).
```

Comment:
It might be worth mentioning the famous https://github.com/jaor/xmobar/issues/239#issuecomment-233206552 which allows trayer to be seamlessly embedded into xmobar

Reply:
I was thinking about this and I ultimately decided that it would perhaps be too confusing for a beginner.  But then again, people may be much more familiar with shell scripting than with Haskell so it may not be a big deal after all

Reply:
@TheMC47 I added a little note in [this](https://github.com/xmonad/xmonad/pull/278/commits/e6c2a670dda3504209a359ccf7cff9992fe01c37) fixup commit; do you think that's enough or should I go into a little bit more detail?

Reply:
I think it's fine :+1: